src/Core/Domain/Product/ValueObject/Condition.php

```diff
@@ -0,0 +1,97 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+
+namespace PrestaShop\PrestaShop\Core\Domain\Product\ValueObject;
+
+use PrestaShop\PrestaShop\Core\Domain\Product\Exception\ProductConstraintException;
+
+/**
+ * Holds product condition either its new , used etc...
+ */
+final class Condition
+{
+    public const IS_NEW = 'new';
+
+    public const IS_USED = 'used';
+
+    public const IS_REFURBISHED = 'refurbished';
+
+    public const AVAILABLE_CONDITIONS = [
+        self::IS_NEW,
+        self::IS_USED,
+        self::IS_REFURBISHED,
+    ];
+
+    /**
+     * @var string
+     */
+    private $condition;
+
+    /**
+     * @var bool
+     */
+    private $displayedOnProductPage;
+
+    /**
+     * @param string $condition
+     * @param bool $displayedOnProductPage
+     *
+     * @throws ProductConstraintException
+     */
+    public function __construct(string $condition, bool $displayedOnProductPage)
```

Comment:
Is `$displayedOnProductPage` really part of condition?

Reply:
there's a tiny checkbox in the form  to mark such thing - either it is visible in the product page or not.

Reply:
Yeah, but why it is related to condition? :thinking:

Reply:
do you have suggestions how I can move out such things? They both are related in the form - I dont want have million setters here - there are currently 20 setters already in the command which targers Product ObjectModel. Do you suggest to put another separate parameter `isConditionDisplayedOnProductPage` or something?

Reply:
For me, separate property would be the best. See Pablo's answer about using setters, this would work imo.